src/tasks/bundler.rs

```diff
@@ -61,42 +89,28 @@ impl BundlePoller {
             .error_for_status()?;
 
         let body = result.bytes().await?;
-        let bundles: TxPoolBundleResponse = serde_json::from_slice(&body)?;
-
-        bundles.bundles.iter().for_each(|bundle| {
-            self.check_seen_bundles(bundle.clone(), &mut unique);
-        });
+        let resp: TxPoolBundleResponse = serde_json::from_slice(&body)?;
 
-        Ok(unique)
+        Ok(resp.bundles)
     }
 
-    /// Checks if the bundle has been seen before and if not, adds it to the unique bundles list.
-    fn check_seen_bundles(&mut self, bundle: Bundle, unique: &mut Vec<Bundle>) {
-        self.seen_uuids.entry(bundle.id.clone()).or_insert_with(|| {
-            // add to the set of unique bundles
-            unique.push(bundle.clone());
-            Instant::now() + Duration::from_secs(self.config.tx_pool_cache_duration)
-        });
-    }
-
-    /// Evicts expired bundles from the cache.
-    pub fn evict(&mut self) {
-        let expired_keys: Vec<String> = self
-            .seen_uuids
-            .iter()
-            .filter_map(
-                |(key, expiry)| {
-                    if expiry.elapsed().is_zero() {
-                        Some(key.clone())
-                    } else {
-                        None
+    /// Spawns a task that simply sends out any bundles it ever finds
+    pub fn spawn(mut self) -> (UnboundedReceiver<Arc<Bundle>>, JoinHandle<()>) {
+        let (outbound, inbound) = unbounded_channel();
+        let jh = tokio::spawn(async move {
+            loop {
+                if let Ok(bundles) = self.check_bundle_cache().await {
+                    tracing::debug!(count = ?bundles.len(), "found bundles");
+                    for bundle in bundles.iter() {
```

Comment:
if you use `into_iter` you'll get an iterator over owned values, and don't need the `.clone()`